manager/controllers/utils.go

```diff
@@ -237,17 +237,19 @@ func ApplyTagsFilter(c *gin.Context, tx *gorm.DB, systemIDExpr string) (*gorm.DB
 	if !enableCyndiTags {
 		return tx, false
 	}
+	var applied bool
 
 	subq := database.Db.
 		Table("inventory.hosts h").
-		Select("h.id::text")
+		Select("h.id")
 
 	tags := c.QueryArray("tags")
 	if len(tags) > 0 {
 		for _, t := range tags {
 			matches := tagRegex.FindStringSubmatch(t)
 			tagJSON := fmt.Sprintf(`[{"namespace":"%s", "key": "%s", "value": "%s"}]`, matches[1], matches[2], matches[3])
 			subq = subq.Where(" h.tags @> ?::jsonb", tagJSON)
+			applied = true
```

Comment:
applied = true can go outside the for loop (thus be assigned just once)

Reply:
Resolved, we don't even need the if check